### PR TITLE
feat: Register, login, logout, verify token works on happy path. Edge cases handling for register and login done.

### PR DESCRIPTION
**Pull request by Drajad-Kusuma-Adi**
N.B: Edge case for logout and verify token is minimal, so I didn't write a test for them.

## Summary by CodeRabbit

- **New Features**
	- Introduced user authentication capabilities including registration, login, and logout.
	- Added API routes for login, authentication verification, and logout.
- **Enhancements**
	- Improved user authentication with token generation and validation.
	- Updated user model to enhance security by changing visibility of certain attributes.
- **Bug Fixes**
	- Adjusted user field naming in database seeding for consistency.
- **Tests**
	- Expanded authentication tests to cover both typical scenarios and edge cases.

### CODE REVIEW

File: tests/Feature/AuthTest.php [hunk 1]
```diff
@@ -1,30 +1,85 @@
...
+        $this->assertStringContainsString('Login berhasil', $loginResponse->getContent());
+        $token = json_decode($registerResponse->getContent())[0]->user->remember_token;
+
+        // Verify the user's authentication
+        $verifyResponse = $this->withHeaders([
+            'Accept' => 'application/json',
+            'Authorization' => 'Bearer '. $token
+        ])->post($this->baseUrl.'/auth/verifyauth');
+        $this->assertEquals(200, $verifyResponse->getStatusCode());
+        $this->assertStringContainsString('Token terverifikasi', $verifyResponse->getContent());
```
[the remark this answers is not in the document]
Comment (Drajad-Kusuma-Adi) [reply tests/Feature/AuthTest.php:52]: Tests for registration, login, logout, and token verification are condensed into one single test named "test_happy_path".

**Review by Drajad-Kusuma-Adi — COMMENTED** [review]
LGTM
